### PR TITLE
Test concurrent writer update file metrics [databricks]

### LINKED ISSUE #8533: [AUDIT][SPARK-43281] Fix concurrent writer does not update file metrics
BasicWriteStatsTaskTracker was updated in https://github.com/apache/spark/commit/ee12c81cf3 to fix a bug, and BasicColumnarWriteStatsTaskTracker has similar code.  We probably need to update it and GpuFileFormatDataWriter to pick up the same fix.

### PR DESCRIPTION
**Pull request by thirtiseven**
Closes #8533 

BasicWriteStatsTaskTracker was updated in https://github.com/apache/spark/commit/ee12c81cf3 to fix a bug, and BasicColumnarWriteTaskStatsTracker has similar code. 

I followed the test in the commit, and it can passed. 

It is because we already have code similar to `statsTrackers.foreach(_.closeFile(status.outputWriter.path()))` in https://github.com/NVIDIA/spark-rapids/blob/ebf852530e6af9c186d974d6c38500608790604c/sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuFileFormatDataWriter.scala#L689

And if this line is deleted, the test added in this PR will failed. 

So I think it is safe to follow the Spark commit to clean up the code which use temp file path to get file status after commit task.

### CODE REVIEW

**Comment by thirtiseven** [comment]
build

**Comment by thirtiseven** [comment]
build

File: tests/src/test/spark321/scala/com/nvidia/spark/rapids/ConcurrentWriterMetricsSuite.scala [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*** spark-rapids-shim-json-lines
+{"spark": "321"}
```
Comment (jlowe) [inline tests/src/test/spark321/scala/com/nvidia/spark/rapids/ConcurrentWriterMetricsSuite.scala:18]: What about Cloudera (cdh) and Databricks (db) shims?

**Comment by thirtiseven** [comment]
build

File: tests/src/test/spark321/scala/com/nvidia/spark/rapids/ConcurrentWriterMetricsSuite.scala (hunk 1 above)
Comment (thirtiseven) [reply tests/src/test/spark321/scala/com/nvidia/spark/rapids/ConcurrentWriterMetricsSuite.scala]: Updated.

**Review by jlowe — APPROVED** [review]

**Pull request merged**